numpy/linalg/linalg.py

```diff
@@ -2788,10 +2795,11 @@ def _multi_dot_matrix_chain_order(arrays, return_costs=False):
     return (s, m) if return_costs else s
 
 
-def _multi_dot(arrays, order, i, j):
+def _multi_dot(arrays, order, i, j, out=None):
     """Actually do the multiplication with the given order."""
     if i == j:
         return arrays[i]
```

Comment:
```suggestion
        assert out is None  # the initial call should never get here
        return arrays[i]
```

Reply:
added in 80c3034cff020f4c6af542f6c9f7a3bcea598319